tests/mocks.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python3.4
+#
+# Copyright 2016 Google Inc.
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This module has common mock objects and functions used in Mobly unit tests.
+
+import logging
+import mock
+import os
+
+
+def get_mock_ads(num):
+    """Generates a list of mock AndroidDevice objects.
+
+    The serial number of each device will be integer 0 through num - 1.
+
+    Args:
+        num: An integer that is the number of mock AndroidDevice objects to
+            create.
+    """
+    ads = []
+    for i in range(num):
+        ad = mock.MagicMock(name="AndroidDevice", serial=i, h_port=None)
+        ads.append(ad)
+    return ads
+
+
+def mock_get_all_instances():
```

Comment:
It is not the full story since it has mocks for adb and fastboot, which are needed for android_device tests but adb and fastboot proxies are not actually part of android_device